src/OpenFeature.DependencyInjection/OpenFeatureServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,53 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using OpenFeature.Internal;
+using OpenFeature.Model;
+
+namespace OpenFeature;
+
+/// <summary>
+/// Contains extension methods for the <see cref="IServiceCollection"/> class.
+/// </summary>
+public static class OpenFeatureServiceCollectionExtensions
+{
+    /// <summary>
+    /// This method is used to add OpenFeature to the service collection.
+    /// OpenFeature will be registered as a singleton.
+    /// </summary>
+    /// <param name="services"><see cref="IServiceCollection"/></param>
+    /// <param name="configure">the desired configuration</param>
+    /// <returns>the current <see cref="IServiceCollection"/> instance</returns>
+    public static IServiceCollection AddOpenFeature(this IServiceCollection services, Action<OpenFeatureBuilder> configure)
+    {
+        Guard.ThrowIfNull(services);
+        Guard.ThrowIfNull(configure);
+
+        services.TryAddSingleton(Api.Instance);
+        services.TryAddSingleton<IFeatureLifecycleManager, FeatureLifecycleManager>();
+
+        var builder = new OpenFeatureBuilder(services);
+        configure(builder);
+
+        if (builder.IsContextConfigured)
+        {
+            services.TryAddScoped<IFeatureClient>(static provider =>
+            {
+                var api = provider.GetRequiredService<Api>();
+                var client = api.GetClient();
+                var context = provider.GetRequiredService<EvaluationContext>();
+                client.SetContext(context);
```

Comment:
Context can be added at both the [global](https://github.com/open-feature/dotnet-sdk/blob/c471c062cf70d78b67f597f468c62dbfbf0674d2/README.md?plain=1#L112) and client level. Global context is most useful for static values (for example, a value representing the timezone of the server or the cloud provider it's running on, or the application version).

We might need multiple contexts setting methods for the different scopes: global, client, and perhaps transaction level once we implement the [transaction propagation](https://github.com/open-feature/dotnet-sdk/issues/243) feature. As it is, I'm not sure it's obvious to a user which one would be used when they do `AddContext(...)`